crates/turbopack-ecmascript/tests/tree-shaker/analyzer/failed-1/output.md

```diff
@@ -0,0 +1,477 @@
+
+```
+## Part 4
+```js
+let source;
+export { source };
+
+```
+## Part 5
+```js
+const eventCallbacks = [];
+export { eventCallbacks };
+
+```
+## Merged (module eval)
+```js
+"module evaluation";
+
+```
+# Entrypoints
+
+```
+{
+    ModuleEvaluation: 0,
+    Export(
+        "connectHMR",
+    ): 3,
+    Export(
+        "addMessageListener",
+    ): 1,
+    Export(
+        "sendMessage",
+    ): 2,
+}
+```
+
+
+# Modules (prod)
+## Part 0
+```js
+"module evaluation";
+
+```
+## Part 1
+```js
+import { eventCallbacks } from "__TURBOPACK_PART__" assert {
+    __turbopack_part__: 5
+};
+export { addMessageListener };
+function addMessageListener(cb) {
+    eventCallbacks.push(cb);
+}
+export { addMessageListener };
+
+```
+## Part 2
+```js
+import { source } from "__TURBOPACK_PART__" assert {
+    __turbopack_part__: 4
+};
+export { sendMessage };
+function sendMessage(data) {
+    if (!source || source.readyState !== source.OPEN) return;
+    return source.send(data);
+}
+export { sendMessage };
+
+```
+## Part 3
+```js
+import { source } from "__TURBOPACK_PART__" assert {
+    __turbopack_part__: 4
+};
+import { eventCallbacks } from "__TURBOPACK_PART__" assert {
+    __turbopack_part__: 5
+};
+export { connectHMR };
+function getSocketProtocol(assetPrefix) {
+    let protocol = location.protocol;
+    try {
+        protocol = new URL(assetPrefix).protocol;
+    } catch (_) {}
+    return protocol === "http:" ? "ws" : "wss";
+}
+function connectHMR(options) {
+    const { timeout = 5 * 1000 } = options;
+    function init() {
+        if (source) source.close();
+        console.log("[HMR] connecting...");
+        function handleOnline() {
+            const connected = {
+                type: "turbopack-connected"
+            };
+            eventCallbacks.forEach((cb)=>{
+                cb(connected);
+            });
+            if (options.log) console.log("[HMR] connected");
+        }
+        function handleMessage(event) {
+            const message = {
+                type: "turbopack-message",
+                data: JSON.parse(event.data)
+            };
+            eventCallbacks.forEach((cb)=>{
+                cb(message);
+            });
+        }
+        function handleDisconnect() {
+            source.close();
+            setTimeout(init, timeout);
+        }
+        const { hostname, port } = location;
+        const protocol = getSocketProtocol(options.assetPrefix || "");
+        const assetPrefix = options.assetPrefix.replace(/^\/+/, "");
+        let url = `${protocol}://${hostname}:${port}${assetPrefix ? `/${assetPrefix}` : ""}`;
+        if (assetPrefix.startsWith("http")) {
+            url = `${protocol}://${assetPrefix.split("://")[1]}`;
+        }
+        source = new window.WebSocket(`${url}${options.path}`);
+        source.onopen = handleOnline;
+        source.onerror = handleDisconnect;
+        source.onmessage = handleMessage;
+    }
+    init();
+}
+export { getSocketProtocol };
```

Comment:
This shouldn't be exported.

Reply:
It's not an export. It's cross-part connector